### PR TITLE
pkp/pkp-lib#3691 default style for HTML galley

### PR DESCRIPTION
**Pull request by Vitaliy-1**

### CODE REVIEW

File: classes/plugins/ThemePlugin.inc.php [hunk 1]
```diff
@@ -821,6 +824,57 @@ function isColourDark( $colour, $limit = 130 ) {
...
+		
+		$submissionFile = $galley->getFile();
+		
+		// Check if HTML galley has attached styles
+		$submissionFileDao = DAORegistry::getDAO('SubmissionFileDAO');
+		import('lib.pkp.classes.submission.SubmissionFile');
+		$embeddableFiles = array_merge(
+			$submissionFileDao->getLatestRevisions($submissionFile->getSubmissionId(), SUBMISSION_FILE_PROOF),
+			$submissionFileDao->getLatestRevisionsByAssocId(ASSOC_TYPE_SUBMISSION_FILE, $submissionFile->getFileId(), $submissionFile->getSubmissionId(), SUBMISSION_FILE_DEPENDENT)
+		);
```
Comment (NateWr) [inline classes/plugins/ThemePlugin.inc.php]: If you add the `$embeddableFiles` to the hook in the OJS commit, you can avoid these requests to the DB.

File: classes/plugins/ThemePlugin.inc.php [hunk 2]
```diff
@@ -821,6 +824,57 @@ function isColourDark( $colour, $limit = 130 ) {
...
+		foreach ($embeddableFiles as $embeddableFile) {
+			if ($embeddableFile->getFileType() =='text/css') $attachedStyles = true;
+		}
+		
+		if ($attachedStyles) return false;
+		
+		// Upload styles to the HTML galley
+		$output = '';
+		
+		foreach ($this->styles as $styleData) {
```
Comment (NateWr) [inline classes/plugins/ThemePlugin.inc.php]: Where do these come from? I'm confused about how a `ThemePlugin` will register a default CSS file. Also, in this case, a LESS file can't be used. 

Is there any way we can use our LESS compiler to support a LESS file as well? I imagine I'd want to reuse the variables from my theme's LESS stylesheets in the HTML galley as well.

Comment (Vitaliy-1) [reply classes/plugins/ThemePlugin.inc.php]: Styles in the `ThemePlugin` are passing to the `PKPTemplateManager` through `_registerStyles` method but here I'm retrieving them directly as they actually won't go to the template; they are injected directly into the HTML. Probably, I need to ensure they that are beeing removed from `$styles` property after beeing used and not passing further to the Template Manager. Or you have something different in mind?

I think it is not a problem to add a call to the compile library in t his method for LESS files if the overall approach looks reasonable.

Comment (NateWr) [reply classes/plugins/ThemePlugin.inc.php]: Ah ok, so if I'm reading this properly, a stylesheet is added in a theme like this:

```
$this->addStyle('name', 'path/to/file.css', ['contexts' => 'htmlGalley']);
```

Is that right? If so, then LESS files should already be handled. `_registerStyles` will transform the URL appropriately.

You'll need to handle the case where `contexts` is an array. `TemplateManager::getResourcesByContext($resources, $context)` can do this for you. Usage would be like this:

```
$templateMgr = TemplateManager::getManager($request);
$styles = $templateMgr->getResourcesByContext($this->styles, 'htmlGalley');
```

That will get styles with any context that includes `htmlGalley`. :+1:

Comment (Vitaliy-1) [reply classes/plugins/ThemePlugin.inc.php]: Yes, right. But the method retrieves the data from styles earlier, before they are registered and passed to Template Manager's methods.

File: classes/template/PKPTemplateManager.inc.php [hunk 3]
```diff
@@ -1452,6 +1455,48 @@ function smartyLoadStylesheet($params, $smarty) {
...
+		if (!empty($styles)) {
+			foreach ($styles as $htmlStyles) {
+				foreach ($htmlStyles as $htmlStyle) {
+					$output .= '<link rel="stylesheet" href="' . $htmlStyle['style'] . '" type="text/css" />' . "\n";
+				}
+			}
+		}
+		
+		if (!empty($output)) {
+			$contents = str_replace('<head>', '<head>' . "\n" . $output, $contents);
```
Comment (NateWr) [inline classes/template/PKPTemplateManager.inc.php]: Let's try `str_ireplace` so it's case-insensitive. We'll probably pick up a few more cases that way.

**Comment by NateWr** [comment]
Replaced by https://github.com/pkp/pkp-lib/pull/4344

**Pull request closed without merging**